### PR TITLE
Uses a PacketQueue for packets coming out of the SCTP stack in order to

### PR DESCRIPTION
**Pull request by bgrozev**
ensure that the calling thread does not block. We have
observed that blocking this thread (e.g. by having it eventually try
to write to a Socket) results in calls to usrsctp_socket to block (i.e.
to our inability to initialize any new SCTP sockets).

### CODE REVIEW

**Comment by brianh5** [comment]
could this potentially result in long datachannel connection times (~30 seconds)?  we've seen that happen on occasion...not sure if that's just the nature of sctp or if it may be related to something like this.

**Comment by bgrozev** [comment]
Possibly, but I think you would have noticed other problems and not just slow SCTP establishment. In what we observed, while the thread was blocked health checks never returned, at least some COLIBRI requests never returned and the statistics were broken.

File: src/main/java/org/jitsi/videobridge/SctpConnection.java [hunk 1]
```diff
@@ -195,6 +196,42 @@ private static synchronized int generateDebugId()
...
+        @Override
+        public boolean handlePacket(RawPacket pkt)
+        {
+            if (pkt == null)
+                return true;
+
+            if (transformer == null)
+                return false;
+
+            System.err.println("XXX sending sctp data over DTLS");
```
Comment (paweldomas) [inline src/main/java/org/jitsi/videobridge/SctpConnection.java]: System.err.println ? Forgot to remove before commit ?

File: src/main/java/org/jitsi/videobridge/SctpConnection.java [hunk 2]
```diff
@@ -195,6 +196,42 @@ private static synchronized int generateDebugId()
...
+     * SCTP packets.
+     */
+    private DtlsPacketTransformer transformer = null;
+
+    /**
+     * The instance which we use to handle packets read from
+     * {@link #packetQueue}.
+     */
+    private final PacketQueue.PacketHandler<RawPacket> handler
+        = new PacketQueue.PacketHandler<RawPacket>()
```
Comment (paweldomas) [inline src/main/java/org/jitsi/videobridge/SctpConnection.java]: Don't really like creating anonymous class here, but that's just my opinion. Would prefer to have named class at the bottom, does the same thing.

**Pull request merged**